clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -2469,11 +2469,20 @@ static bool CheckConstexprFunctionBody(Sema &SemaRef, const FunctionDecl *Dcl,
   //     base class sub-objects shall be a constexpr constructor.
   //
   // Note that this rule is distinct from the "requirements for a constexpr
-  // function", so is not checked in CheckValid mode.
+  // function", so is not checked in CheckValid mode. Because the check for
+  // constexpr potential is expensive, skip the check if the diagnostic is
+  // disabled, the function is declared in a system header, or we're in C++23
+  // or later mode (see https://wg21.link/P2448).
+  auto SkipCheck = [&SemaRef, Dcl] {
```

Comment:
Is lambda necessary here?

Reply:
Is that not to check all this when `Kind != Sema::CheckConstexprKind::Diagnose`? I was under impression that `Expr::isPotentialConstantExpr` is the expensive call.

Reply:
This is done now.